### DISCUSSION TITLE
Fixtures support in JUnit 5

### DISCUSSION THREAD

**Discussion by gnl42** [eclipse-mylyn/org.eclipse.mylyn#595] [General]
As @BeckerFrank noted in #589, fixtures need to be supported in JUnit5.
The problem is that JUnit 5 doesn't have anything equivalent to the TestSuite used in Junit4 to pragmatically create a test suite for each fixture found. 

I'm not sure what a good solution would be:

1. Only have one version of each server, ie. not allow multiple fixtures:
    The current multipass setup only has one of each type (should Bugzilla 5 be added to the mix?)
    
2. Dig into the depths of JUnit5 and come up with a TestSuite framework:
    Not a fan of the idea. Would add complexity to the test support code we don't need.
    And should really live with the JUnit code base, not Mylyn
    
3. Use something like @MethodSource for all appropriate fixtures:
    Would make for strange reading of the test runs. Possibly mitigated by used the ordered method option

4. Create AllXXXTests suite for each fixture. E.g. AllBugzilla4Tests, AllBugzilla5Tests:
    I can see it using combination of Extensions, Annotations, and maybe ArgumentsProvider  for the tests in a class
    
    Not sure how you would run specific tests against a specific fixture?
    
5. ???

Comments?

Thanks
 George

**Comment by BeckerFrank** [comment]
I create a POC in branch [POC_Junit5_BugzillaRest](https://github.com/eclipse-mylyn/org.eclipse.mylyn/tree/POC_Junit5_BugzillaRest).
